modules/ting/src/Search/TingSearchResultInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * @file
+ * The TingSearchResultInterface interface.
+ */
+
+namespace Ting\Search;
+
+/**
+ * Interface TingSearchResultInterface
+ *
+ * @package Ting\Search
+ */
+interface TingSearchResultInterface {
```

Comment:
Did consider ArrayAccess - opted for delaying that decision until I got to some some more iteration of the result object.
In the same vain: For now I've allowed some of the open search naming to leak trough, with the intention of refactoring when I get to the actual primo implementation.

That being said you have a valid point with regards to the counting - I'll give it a look.

With regards to getCollections() - both the search-providers and Drupal has the concept of "Collections". The first is wrappers around somewhat raw search-results and the latter is an entity. It's valid to call both "Collections". Which is why I went for being explicit and calling the getter getTingEntityCollections.

Sidenote: shouldn't the TingCollection Class in ding which is a subclass of DingEntityCollection have been called TingEntityCollection?